### PR TITLE
Added option to ignore regexp-matching users by scheduler.

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: Added option for blocksconvert scheduler to ignore users based on regexp.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: tools/blocksconvert/scheduler/scheduler.go [hunk 1]
```diff
@@ -51,19 +54,29 @@ func NewScheduler(cfg Config, scfg blocksconvert.SharedConfig, l log.Logger, reg
 		users = blocksconvert.ParseAllowedUsers(cfg.AllowedUsers)
 	}
 
-	s := newSchedulerWithBucket(l, b, scfg.BucketPrefix, users, cfg, reg)
+	var ignoredUserRegex *regexp.Regexp = nil
+	if users.AllUsersAllowed() && cfg.IgnoredUserPattern != "" {
```
Comment (pracucci) [inline tools/blocksconvert/scheduler/scheduler.go:58]: Why not applying ignored users even if `users.AllUsersAllowed()` is false?

**Review by pracucci — APPROVED** [review]

File: tools/blocksconvert/scheduler/scheduler.go (hunk 1 above)
Comment (pstibrany) [reply tools/blocksconvert/scheduler/scheduler.go]: You're right. I copied the logic from scanner, where these two options are exclusive. It works slightly differently here, although I would only recommend using one option.

Comment (pstibrany) [reply tools/blocksconvert/scheduler/scheduler.go]: I've modified scanner to have the same logic wrt. to options as scheduler. (Internally the logic was already the same, just config options were treated differently).

Comment (pracucci) [reply tools/blocksconvert/scheduler/scheduler.go]: Ok. Makes sense to me.

**Review by pracucci — APPROVED** [review]
Still LGTM 😉  I think we can merge!

**Pull request merged**